### DISCUSSION TITLE
mgr prometheus module fails with rook 1.13.1 and ceph 18.2.1

### DISCUSSION THREAD

**Discussion by arichard42** [rook/rook#13481] [General]
My test cluster was upgraded using helm chart 1.13.1 and ceph 18.2.1.

Since then, the mgr prometheus module fails to load, so the prometheus serviceMonitor/rook-ceph/rook-ceph-mgr target is down as the corresponding port (9283) is not listening on the active mgr pod although the mgr module prometheus is marked as active.

Looking at the logs (using "ceph module disable prometheus" followed by "ceph module enable prometheus") shows :

debug 2023-12-29T14:01:00.484+0000 7fcb4aba0700  0 [prometheus DEBUG root] setting log level based on debug_mgr: INFO (2/5)
debug 2023-12-29T14:01:00.485+0000 7fcb4aba0700 -1 no module 'rook'
debug 2023-12-29T14:01:00.485+0000 7fcb4aba0700 -1 no module 'rook'
debug 2023-12-29T14:01:00.485+0000 7fcb4aba0700 -1 mgr load Failed to construct class in 'prometheus'
debug 2023-12-29T14:01:00.486+0000 7fcb2207c700  0 [progress INFO root] Loading...
debug 2023-12-29T14:01:00.486+0000 7fcb2207c700  0 [progress INFO root] Loaded [<progress.module.GhostEvent object at 0x7fcb2e2bef28>, <progress.module.GhostEvent object at 0x7fcb2e2bef60>, <progress.module.GhostEvent object at 0x7fcb2e2bef98>, <progress.module.GhostEvent object at 0x7fcb2e2befd0>, <progress.module.GhostEvent object at 0x7fcb2e2cf048>, <progress.module.GhostEvent object at 0x7fcb2e2cf080>, <progress.module.GhostEvent object at 0x7fcb2e2cf0b8>, <progress.module.GhostEvent object at 0x7fcb2e2cf0f0>, <progress.module.GhostEvent object at 0x7fcb2e2cf128>, <progress.module.GhostEvent object at 0x7fcb2e2cf160>, <progress.module.GhostEvent object at 0x7fcb2e2cf198>, <progress.module.GhostEvent object at 0x7fcb2e2cf1d0>, <progress.module.GhostEvent object at 0x7fcb2e2cf208>, <progress.module.GhostEvent object at 0x7fcb2e2cf240>, <progress.module.GhostEvent object at 0x7fcb2e2cf278>, <progress.module.GhostEvent object at 0x7fcb2e2cf2b0>, <progress.module.GhostEvent object at 0x7fcb2e2cf2e8>, <progress.module.GhostEvent object at 0x7fcb2e2cf320>, <progress.module.GhostEvent object at 0x7fcb2e2cf358>, <progress.module.GhostEvent object at 0x7fcb2e2cf390>, <progress.module.GhostEvent object at 0x7fcb2e2cf3c8>, <progress.module.GhostEvent object at 0x7fcb2e2cf400>, <progress.module.GhostEvent object at 0x7fcb2e2cf438>, <progress.module.GhostEvent object at 0x7fcb2e2cf470>, <progress.module.GhostEvent object at 0x7fcb2e2cf4a8>, <progress.module.GhostEvent object at 0x7fcb2e2cf4e0>, <progress.module.GhostEvent object at 0x7fcb2e2cf518>, <progress.module.GhostEvent object at 0x7fcb2e2cf550>, <progress.module.GhostEvent object at 0x7fcb2e2cf588>, <progress.module.GhostEvent object at 0x7fcb2e2cf5c0>, <progress.module.GhostEvent object at 0x7fcb2e2cf5f8>, <progress.module.GhostEvent object at 0x7fcb2e2cf630>, <progress.module.GhostEvent object at 0x7fcb2e2cf668>] historic events
debug 2023-12-29T14:01:00.486+0000 7fcb2207c700  0 [progress INFO root] Loaded OSDMap, ready.
debug 2023-12-29T14:01:00.491+0000 7fcb2307e700  0 [pg_autoscaler INFO root] _maybe_adjust
debug 2023-12-29T14:01:00.485+0000 7fcb4aba0700 -1 mgr load Traceback (most recent call last):
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 1657, in _oremote
    return mgr.remote(o, meth, *args, **kwargs)
  File "/usr/share/ceph/mgr/mgr_module.py", line 2228, in remote
    args, kwargs)
ImportError: Module not found

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/share/ceph/mgr/prometheus/module.py", line 649, in __init__
    self.modify_instance_id = self.get_orch_status() and self.get_module_option(
  File "/usr/share/ceph/mgr/prometheus/module.py", line 869, in get_orch_status
    return self.available()[0]
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 1586, in inner
    completion = self._oremote(method_name, args, kwargs)
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 1661, in _oremote
    f_set = self.get_feature_set()
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 1586, in inner
    completion = self._oremote(method_name, args, kwargs)
  File "/usr/share/ceph/mgr/orchestrator/_interface.py", line 1657, in _oremote
    return mgr.remote(o, meth, *args, **kwargs)
  File "/usr/share/ceph/mgr/mgr_module.py", line 2228, in remote
    args, kwargs)
ImportError: Module not found

debug 2023-12-29T14:01:00.507+0000 7fcb4aba0700 -1 mgr operator() Failed to run module in active mode ('prometheus')

This problem was not present with ceph 18.2.0.

**Comment by travisn** [comment]
@avanthakkar could you take a look?

**Reply by avanthakkar** [reply]
Looking at traceback it seems like probably some issu coming from orchestration/rook modul. Probably @jmolmo mayb know better on this

**Reply by travisn** [reply]
@jmolmo Can you take a look?

**Comment by ebCrypto** [comment]
I have the same problem `mgr operator() Failed to run module in active mode ('prometheus')` with ceph 18.2.1. Downgrading to 18.2.0 fixed the issue. Thank you @arichard42 for pointing this out.

**Comment by jameshearttech** [comment]
We are using Rook v1.13.1. We upgraded from 17.x to 18.2.1 to fix the issue where the dashboard is unable to connect to object store gateway. Here is the [link to the thread in Slack](https://rook-io.slack.com/archives/CK9CF5H2R/p1703837457031159) where I was rambling about it. I had not tested 18.2.0 because I read the issue is fixed in 18.2.1 in [#12099](https://github.com/rook/rook/issues/12099). I just tested 18.2.0 and confirmed the dashboard issue is present in that version.  For the time being we must choose between Prometheus metrics or dashboard / object gateway integration (e.g., check bucket usage).
![image](https://github.com/rook/rook/assets/67252279/68427068-eb18-4f8c-b43d-673ff4187cb0)

**Reply by ebCrypto** [reply]
Same here on 18.2.0. `The Object Gateway Service is not configured`

**Reply by rkachach** [reply]
@jameshearttech @ebCrypto This is another issue and is not related to the discussion. Please, create a new one. As of the issue you are observing it was fixed in `v18.2.1` which comes with `Rook v1.13.1`

**Comment by rkachach** [comment]
This is being tracked by https://github.com/rook/rook/issues/13527. Please, use the issue tracker to add new information to keep one discussion thread only.

**Comment by jameshearttech** [comment]
> @jameshearttech @ebCrypto This is another issue and is not related to the discussion. Please, create a new one. As of the issue you are observing it was fixed in `v18.2.1` which comes with `Rook v1.13.1`

The default image tag for Ceph in Rook v1.13.1 is v17.2.6 not v18.2.1; however, Rook v1.13.1 supports Ceph v18.x. This is important to note for anyone reading this thread.

```
$ helm show values rook-ceph-cluster-v1.13.1.tgz | grep 'cephVersion:' -B 1 -A 7
  # For more details, check https://rook.io/docs/rook/v1.10/CRDs/Cluster/ceph-cluster-crd/
  cephVersion:
    # The container image used to launch the Ceph daemon pods (mon, mgr, osd, mds, rgw).
    # v17 is Quincy, v18 is Reef.
    # RECOMMENDATION: In production, use a specific version tag instead of the general v18 flag, which pulls the latest release and could result in different
    # versions running within the cluster. See tags available at https://hub.docker.com/r/ceph/ceph/tags/.
    # If you want to be more precise, you can always use a timestamp tag such as quay.io/ceph/ceph:v17.2.6-20230410
    # This tag might not contain a new Ceph version, just security fixes from the underlying operating system, which will reduce vulnerabilities
    image: quay.io/ceph/ceph:v17.2.6
```

**Reply by Jean-Daniel** [reply]
The default tag for rook 1.13.2 is 18.2.1, and the issue is still present.

**Comment by arichard42** [comment]
I confirm the patch v1 proposed on #13527 fix the issue for me.
